src/kaleidoscope/device/keyboardio/Imago.cpp

```diff
@@ -0,0 +1,191 @@
+/* -*- mode: c++ -*-
+ * Kaleidoscope-Hardware-Keyboardio-Imago -- Imago hardware support for Kaleidoscope
+ * Copyright (C) 2018, 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of version 3 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef ARDUINO_AVR_KEYBOARDIO_IMAGO
+
+#include <Kaleidoscope.h>
+
+extern "C" {
+#include "twi.h"
+}
+
+#define ELEMENTS(arr)  (sizeof(arr) / sizeof((arr)[0]))
+
+#define LED_DRIVER_ADDR 0x30
+
+namespace kaleidoscope {
+namespace device {
+namespace keyboardio {
+
+static constexpr uint8_t CMD_SET_REGISTER = 0xFD;
```

Comment:
@algernon Can you open a ticket to lift this LED driver to its own driver class sometime in the future?

Reply:
#709, done